### PR TITLE
[Bug Fix] Catch all recent unsubscribes by updating partitions

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** [dbt_iterable #41](https://github.com/fivetran/dbt_iterable/issues/41)

**This PR will result in the following new package version:** v0.8.1

This new update only impacts customers on the old schema. Minor update instead of breaking change. 

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fixes
- Updated `stg_iterable__user_unsub_message_type` to partition on `message_type_id` along with `email`. This brings in all unsubscriptions for the `unique_user_key` across all message types for an email when the `user_unsubscribed_message_type_history` source table is brought into the staging model.
- Updated `stg_iterable__user_unsubscribed_channel` and `stg_iterable__user_unsubscribed_channel` to partition on `channel_id` along with `email`. This brings in all unsubscription for the `unique_user_key` across all channels for an email  when the `user_unsubscribed_channel_history` source table is brought into the staging model.
- These updates ensure that `iterable__user_unsubscriptions` in the `dbt_iterable` package brings in all the latest unsubscribing behavior for a user by channel and by message type, and not just the last unsubscription for a user based on the `updated_at` value.

## Under the Hood
- Updated seed files to effectively run and test that the new partitions worked as expected.
- Updated [pull request and issue templates](https://github.com/fivetran/dbt_iterable_source/tree/v0.8.1/.github).
- Included auto-releaser GitHub Actions workflow to automate future releases.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

See Height ticket

### If you had to summarize this PR in an emoji, which would it be?
 
🌱

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
+# dbt_iterable_source v0.8.1
+[PR #30](https://github.com/fivetran/dbt_iterable_source/pull/30) introduces the following updates: 
+
+## Bug Fixes
+- Updated `stg_iterable__user_unsub_message_type` to partition on `message_type_id` along with `email`. This brings in all unsubscriptions for the `unique_user_key` across all message types for an email when the `user_unsubscribed_message_type_history` source table is brought into the staging model.
+- Updated `stg_iterable__user_unsubscribed_channel` and `stg_iterable__user_unsubscribed_channel` to partition on `channel_id` along with `email`. This brings in all unsubscription for the `unique_user_key` across all channels for an email  when the `user_unsubscribed_channel_history` source table is brought into the staging model.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: It is reading a bit confusing when you mention that these changes are for when the `user_unsubscribed_message_type_history` or `user_unsubscribed_channel_history` source tables are brought into the staging model. Could we instead mention that these changes are only impacting connectors setup before August 2023?

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-avinash, the code generally looks good to me! I do just have one comment in the CHANGELOG, but nothing which will hold back an approval.

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  call-workflow-passing-data:
+    if: github.event.pull_request.merged
+    uses: fivetran/dbt_package_automations/.github/workflows/auto-release.yml@main
+    secrets: inherit
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply .github/workflows/auto-release.yml:13]: Added!

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md:6]: Committed

**Review by fivetran-reneeli — APPROVED** [review]
lgtm

**Pull request merged**